mlir/include/mlir/Dialect/Affine/IR/AffineOps.td

```diff
@@ -407,7 +407,8 @@ def AffineIfOp : Affine_Op<"if",
     }
     ```
   }];
-  let arguments = (ins Variadic<AnyType>);
+  let arguments = (ins Variadic<AnyType>,
+                       IntegerSetAttr:$condition);
```

Comment:
I don't follow affine dialect but I'm not sure this is the correct way to accomplish what you're trying to accomplish - expose/add default a builder that can successfully construct this op from Python. At minimum this is a breaking change? Though maybe not  because ultimately it's the custom assembly parser that determines things. Anyway you're gonna need to tag someone that works on affine to review these parts.

Reply:
My understanding is AffineIfOp has skipDefaultBuilders=1 which effectively skips the creation of the default AffineIfOp::builder on the C++ side.  (AffineIfOp has two custom OpBuilder defined immediately below in the extraClassDeclaration.)  Interestingly, on the python side, _affine_ops_gen.py shows that the default builder is still being created, but it does not accept IntegerSet and thus makes it useless.  

This fix at line 411 makes the default python AffineIfOp builder take in an IntegerSet input and does impact the C++ side of things.  (This is my understanding and I can be wrong :)   

We definitely need a TD file expert to chime in here!